app/controllers/users_controller.rb

```diff
@@ -217,6 +216,7 @@ def login_user(user)
     store_default_taxonomy(user, 'location') unless session.has_key?(:location_id)
     TopbarSweeper.expire_cache
     telemetry_increment_counter(:successful_ui_logins)
+    user.post_successful_login
```

Comment:
With this change, this gets called twice when logging to the UI as an internal user, once from the depths of `User.try_to_login` on L127 and second time from here by calling `login_user` on L143.

Reply:
After this patch, I also notice that `user.post_successful_login` is also called twice when the external user login successfully for the first time  after auto create user  in `find_or_create_external_user`.

This change seems better?

```
diff --git a/app/models/user.rb b/app/models/user.rb
index a8ca2a1ae..639029c3f 100644
--- a/app/models/user.rb
+++ b/app/models/user.rb
@@ -337,6 +337,7 @@ class User < ApplicationRecord
         user.usergroups = new_usergroups.uniq
       end
 
+      user.post_successful_login
       user
     # not existing user and creating is disabled by settings
     elsif auth_source_name.nil?
```